Realm/RLMRealmConfiguration.mm

```diff
@@ -128,6 +128,8 @@ - (instancetype)copyWithZone:(NSZone *)zone {
     configuration->_migrationBlock = _migrationBlock;
     configuration->_shouldCompactOnLaunch = _shouldCompactOnLaunch;
     configuration->_customSchema = _customSchema;
+    configuration->_eventConfiguration = _eventConfiguration;
+    configuration->_migrationObjectClass = _migrationObjectClass;
```

Comment:
Why do we need to set this `migrationObjectClass` in the configuration?, I see this is set to MigrationObject.self as default and only available to from our Private API, if this is to allow users to setUp their own Migration Object Classes through our private API, or this is for testing purposes?. I don't see this documented anywhere.

Reply:
Inside a migration we need to give the user RLMMigrationObject objects when they're using the obj-c API and MigrationObject objects when they're using the Swift API. The migration code doesn't know which is in use, and even if it did it can't directly reference MigrationObject since that's defined by the Swift framework. Instead we store the class which we want to instantiate in the config, and set it to either the obj-c or swift class depending on how the config is contructed.

Reply:
Ohh I see it, thanks for the explanation